### PR TITLE
Deprecate `nn.glob` package

### PR DESCRIPTION
**Pull request by Padarn**
Removes use of `nn.glob` package. https://github.com/pyg-team/pytorch_geometric/issues/4712

### CODE REVIEW

File: CHANGELOG.md
Comment (lightaime) [inline CHANGELOG.md:40]:
```suggestion
- Added `torch_geometric.nn.aggr` package ([#4687](https://github.com/pyg-team/pytorch_geometric/pull/4687), [#4721](https://github.com/pyg-team/pytorch_geometric/pull/4721), [#4731](https://github.com/pyg-team/pytorch_geometric/pull/4731), [#4762](https://github.com/pyg-team/pytorch_geometric/pull/4762), [#4749](https://github.com/pyg-team/pytorch_geometric/pull/4749), [#4779](https://github.com/pyg-team/pytorch_geometric/pull/4779), [#4863](https://github.com/pyg-team/pytorch_geometric/pull/4863), [#4864](https://github.com/pyg-team/pytorch_geometric/pull/4864), [#4865](https://github.com/pyg-team/pytorch_geometric/pull/4865), [#4866](https://github.com/pyg-team/pytorch_geometric/pull/4866), [#4872](https://github.com/pyg-team/pytorch_geometric/pull/4872), [#4934](https://github.com/pyg-team/pytorch_geometric/pull/4934), [#4935](https://github.com/pyg-team/pytorch_geometric/pull/4935), [#4957](https://github.com/pyg-team/pytorch_geometric/pull/4957), [#4973](https://github.com/pyg-team/pytorch_geometric/pull/4973), [#4973](https://github.com/pyg-team/pytorch_geometric/pull/4973), [#4986](https://github.com/pyg-team/pytorch_geometric/pull/4986), [#4995](https://github.com/pyg-team/pytorch_geometric/pull/4995), [#5000](https://github.com/pyg-team/pytorch_geometric/pull/5000), [#5039](https://github.com/pyg-team/pytorch_geometric/issues/5039)
```

File: torch_geometric/nn/glob/glob.py
Comment (lightaime) [inline torch_geometric/nn/glob/glob.py:80]: Could you please clarify with this is removed completely? If we need it can we implement it with `MultiAggregation`?

File: test/nn/aggr/test_sort.py
Comment (lightaime) [inline test/nn/aggr/test_sort.py:1]: Could you please clarify why the tests are removed?

**Review by lightaime — COMMENTED** [review]
Thanks! LGTM! Left a few minor comments.

File: torch_geometric/nn/pool/glob.py
Comment (Padarn) [reply torch_geometric/nn/pool/glob.py:53]: Thanks good point!

File: torch_geometric/nn/pool/glob.py
Comment (Padarn) [reply torch_geometric/nn/pool/glob.py:19]: Good idea

**Comment by Padarn** [comment]
Thanks for the review @lightaime

File: torch_geometric/nn/glob/glob.py
Comment (Padarn) [reply torch_geometric/nn/glob/glob.py:80]: Yes removed completed (discussed with @rusty1s in earlier version of this PR - the code for  this was added recently and we decided it was no longer needed)

File: test/nn/aggr/test_sort.py
Comment (rusty1s) [inline test/nn/aggr/test_sort.py:6]: Keep this file and rename to `test_sort_aggregation()`?

File: test/nn/glob/test_glob.py
Comment (rusty1s) [inline test/nn/glob/test_glob.py:1]: Move this file to `pool/glob.py`?

File: torch_geometric/nn/pool/__init__.py
Comment (rusty1s) [inline torch_geometric/nn/pool/__init__.py:14]: Ah sorry, this is not what I meant. I meant:
* Move `global_add_pool`, `global_mean_pool`, and `global_max_pool` to `glob/glob.py` to pool/glob.py`.
* Move `glob/__init__.py` to `nn/glob.py` and keep the deprecations in there.

**Review by rusty1s — COMMENTED** [review]
Thanks @Padarn. I think my comments earlier let to some misunderstanding, sorry that. Clarified more in the comments. Let me know what you think!

File: torch_geometric/nn/pool/__init__.py
Comment (rusty1s) [reply torch_geometric/nn/pool/__init__.py:14]: The idea here is that we keep the implementations of `global_*_pool` as they were (since they are so heavily used).

**Comment by Padarn** [comment]
Ah yeah I misunderstood. Thanks for clarifying, I'll make the changes

File: torch_geometric/nn/pool/__init__.py
Comment (Padarn) [reply torch_geometric/nn/pool/__init__.py:14]: hmm sorry I don't fully understand the distinction between these. In both cases you can 
```
from torch_geometric.nn import global_max_pool
```
is it that you want to keep this?
```
from torch_geometric.nn.glob import global_max_pool
````

Comment (Padarn) [reply torch_geometric/nn/pool/__init__.py:14]: Or you just want to clearly separate the deprecations?

Comment (rusty1s) [reply torch_geometric/nn/pool/__init__.py:14]: Yes, we only need to
deprecate the import of `torch_geometric.nn.glob`.

File: test/nn/aggr/test_sort.py
Comment (Padarn) [reply test/nn/aggr/test_sort.py:1]: Mistake, added back.

File: torch_geometric/nn/glob/glob.py
Comment (lightaime) [reply torch_geometric/nn/glob/glob.py:80]: Make sense

File: torch_geometric/nn/glob.py
Comment (lightaime) [inline torch_geometric/nn/glob.py:11]: The deprecations are moved back to `nn.glob.*`. Should we change back from `nn.pool.*` to  `nn.glob.*`? Sorry about that.

**Review by lightaime — APPROVED** [review]
LGTM! Left some small comments.

File: torch_geometric/nn/glob.py
Comment (rusty1s) [reply torch_geometric/nn/glob.py:11]: I think this is correct. We provide backward compatibility in `nn/glob.py` but deprecate its use.

File: torch_geometric/nn/glob.py
Comment (rusty1s) [inline torch_geometric/nn/glob.py:1]: The `global_*_pool` methods need to be added here as well.

File: torch_geometric/nn/pool/__init__.py
Comment (rusty1s) [reply torch_geometric/nn/pool/__init__.py:251]: +1. Let‘s also move the global pooling methods to the top.

File: torch_geometric/nn/pool/glob.py
Comment (rusty1s) [inline torch_geometric/nn/pool/glob.py:32]: Can we leave the implementation as it is? I am not super happy with having global modules here.

**Review by rusty1s — APPROVED** [review]
Thank you! This is exactly what I had in mind. Left some last comments.

**Comment by Padarn** [comment]
Thanks for the reviews again! I've addressed the comments. Will merge later unless any objections

**Comment by Padarn** [comment]
I have some problem with the docs build. Will figure it out tonight

**Comment by Padarn** [comment]
could be https://github.com/sphinx-doc/sphinx/issues/10705 .. may have to wait until they release the new version

**Comment by rusty1s** [comment]
Yes, don‘t worry about it.

**Pull request merged**
